src/V3LinkResolve.cpp

```diff
@@ -76,12 +76,14 @@ class LinkResolveVisitor final : public VNVisitor {
         }
     }
     void visit(AstInitialAutomatic* nodep) override {
-        iterateChildren(nodep);
         // Initial assignments under function/tasks can just be simple
         // assignments without the initial
         if (m_ftaskp) {
             nodep->replaceWith(nodep->stmtsp()->unlinkFrBackWithNext());
             VL_DO_DANGLING(pushDeletep(nodep), nodep);
+        } else {
+            // No need for this if a parent function/task is already iterating
+            iterateChildren(nodep);
```

Comment:
Instead see if works to put at top of function

    if (nodep->user2SetOnce()) return;

Reply:
It doesn't help in `AstInitialAutomatic` visitor, because that one is only hit once.  But it does work in the `AstSFormatF` visitor.